src/CachePeer.cc

```diff
@@ -69,20 +69,11 @@ CachePeer::noteSuccess()
     }
 }
 
-void
-CachePeer::noteFailure(const Http::StatusCode code)
-{
-    if (Http::Is4xx(code))
-        return; // this failure is not our fault
-
-    countFailure();
-}
-
 // TODO: Require callers to detail failures instead of using one (and often
 // misleading!) "connection failed" phrase for all of them.
 /// noteFailure() helper for handling failures attributed to this peer
```

Comment:
FWIW, we are testing changes that address the above TODO. They are independent from this fix.